src/collections/_documentation/platforms/dotnet/nlog.md

```diff
@@ -0,0 +1,169 @@
+
+> NOTE: Messages logged from assemblies with the name starting with `Sentry` will not generate events.
+
+## Features
+
+* Store log messages as breadcrumbs
+* Send events to sentry
+
+Two separate settings define the minimum log level to keep the log entry as a `Breadcrumb` and to send an `Event` to Sentry. The events include any stored breadcrumb on that [scope]({% link _documentation/enriching-error-data/scopes.md %}).
+
+By default, any message with log level `Info` or higher will be kept as a `Breadcrumb`.
+
+The default value to report a log entry as an event to Sentry is `Error`.
+
+This means that out of the box, any `Error` call will create an `Event` which will include all log messages of level `Info`, `Warn` and also `Error` and `Critical`.
+
+
+## Configuration
+
+You can configure the Sentry NLog target via code as follows:
+
+```csharp
+LogManager.Configuration = new LoggingConfiguration();
+LogManager.Configuration
+    .AddSentry(o =>
+    {
+        // Optionally specify a separate format for message
+        o.Layout = "${message}";
+        // Optionally specify a separate format for breadcrumbs
+        o.BreadcrumbLayout = "${logger}: ${message}";
+
+        // Debug and higher are stored as breadcrumbs (default is Info)
+        o.MinimumBreadcrumbLevel = LogLevel.Debug;
+        // Error and higher is sent as event (default is Error)
+        o.MinimumEventLevel = LogLevel.Error;
+
+        // Send the logger name as a tag
+        o.AddTag("logger", "${logger}");
+
+        // All Sentry Options are accessible here.
+    });  
+```
+
+It's also possible to initialize the SDK through the NLog integration (as opposed to using `SentrySdk.Init`). 
+This is useful when the NLog is the only integration being used in your application. To initialize the Sentry SDK through the NLog integration, provide it with the DSN:
+
+```csharp
+LogManager.Configuration = new LoggingConfiguration();
+LogManager.Configuration
+    .AddSentry(o =>
+    {
+        // The NLog integration will initialize the SDK if DSN is set:
+        o.Dsn = new Dsn("___PUBLIC_DSN___"));
+    });  
+```
+
+> NOTE:
+The SDK only needs to be initialized once. If a `DSN` is made available to this integration, by default it **will** initialize the SDK. If you do not wish to initialize the SDK via this integration, set the `InitializeSdk` flag to **false**. Not providing a DSN or leaving it as `null` instructs the integration not to initialize the SDK and unless another integration initializes it or you call `SentrySdk.Init`, the SDK will stay disabled.
+
+The SDK can also be configured via `NLog.config` XML file:
+
+```xml
+<?xml version="1.0" encoding="utf-8" ?>
+<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+>
+  <extensions>
+    <add assembly="Sentry.NLog" />
+  </extensions>
+
+  <targets>
+    <target xsi:type="Sentry" name="sentry"
+            dsn="___PUBLIC_DSN___"
```

Comment:
One thing I just thought of is that I specified that the `Dsn` is specified as a required parameter in the `SentryTarget`, which means that if a sentry target is defined in the `NLog.config` file without a dsn specified, it will probably not register the target correctly. 

I'd have to test it to verify that, but that could cause issues if people want to set the dsn another way. So that should either be documented here or else changed on the target, which I'm happy to do if you think that's the better option.

Reply:
Right. We should make this not required then. Firstly because the lack of `DSN` means _don't initialize Sentry at all_ but also because It's important to make it possible for integration *not* to initialize the SDK. The canonical way to init any Sentry SDK from the unified API is via `Sentry.Init` so allowing an integration initialize the SDK is just a convenience for those who use a single integration. It also helps making things idiomatic.

Could you please modify the integration to make `DSN` optional?

Reply:
Can do. I should be able to send a PR this weekend.